tools/test-proxy/Azure.Sdk.Tools.TestProxy/Models/ActiveMetadataModel.cs

```diff
@@ -1,26 +1,79 @@
-﻿using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.RazorPages;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
 using System.Xml;
 using System.IO;
+using Azure.Sdk.Tools.TestProxy.Common;
+using System.Collections.Concurrent;
+using Microsoft.CodeAnalysis.Operations;
+using Azure.Sdk.Tools.TestProxy.Common.Exceptions;
 
 namespace Azure.Sdk.Tools.TestProxy.Models
 {
     public class ActiveMetadataModel : RunTimeMetaDataModel
     {
+        public ActiveMetadataModel(string recordingId)
+        {
+            RecordingId = recordingId;
+        }
+
         public ActiveMetadataModel(RecordingHandler pageRecordingHandler)
         {
-            Descriptions = _populateFromHandler(pageRecordingHandler);
+            Descriptions = _populateFromHandler(pageRecordingHandler, "");
         }
 
-        private List<ActionDescription> _populateFromHandler(RecordingHandler handler)
+        public ActiveMetadataModel(RecordingHandler pageRecordingHandler, string recordingId)
+        {
+            RecordingId = recordingId;
+            Descriptions = _populateFromHandler(pageRecordingHandler, recordingId);
+        }
+
+        public string RecordingId { get; set; }
+
+        private List<ActionDescription> _populateFromHandler(RecordingHandler handler, string recordingId)
         {
+            var sanitizers = (IEnumerable<RecordedTestSanitizer>) handler.Sanitizers;
+            var transforms = (IEnumerable<ResponseTransform>) handler.Transforms;
+            var matcher = handler.Matcher;
+
+            List<ConcurrentDictionary<string, ModifiableRecordSession>> searchCollections = new List<ConcurrentDictionary<string, ModifiableRecordSession>>()
+            {
+                handler.PlaybackSessions,
+                handler.RecordingSessions,
+                handler.InMemorySessions
+            };
+
+            var recordingFound = false;
+            if (!string.IsNullOrWhiteSpace(recordingId)){
+                foreach (var sessionDict in searchCollections)
+                { 
+                    if (sessionDict.TryGetValue(recordingId, out var session))
+                    {
+                        sanitizers = sanitizers.Concat(session.AdditionalSanitizers);
+                        transforms = transforms.Concat(session.AdditionalTransforms);
+
+                        if (session.CustomMatcher != null)
+                        {
+                            matcher = session.CustomMatcher;
+                        }
+
+                        recordingFound = true;
+                        break;
+                    }
+                }
+
+                if (!recordingFound)
+                {
+                    throw new HttpException(System.Net.HttpStatusCode.BadRequest, $"{recordingId} is not found in any Playback, Recording, or In-Memory sessions.");
```

Comment:
It seems odd to throw api layer exceptions from the model when you're looking for this specific throw in the controller.  i,e. The BadRequest status code and message are unused.

Reply:
Yeah you're right. Converted to `SessionNotActiveException`